pixeltable/share/publish.py

```diff
@@ -102,5 +92,61 @@ def _upload_bundle_to_s3(bundle: Path, parsed_location: urllib.parse.ParseResult
         file=sys.stdout,
     )
     s3_client.upload_file(
-        Filename=str(bundle), Bucket=bucket, Key=str(remote_path), ExtraArgs=upload_args, Callback=progress_bar.update
+        Filename=str(bundle), Bucket=bucket, Key=remote_path, ExtraArgs=upload_args, Callback=progress_bar.update
+    )
+
+
+def pull_replica(dest_path: str, src_tbl_uri: str) -> pxt.Table:
+    headers_json = {'X-api-key': Env.get().pxt_api_key, 'Content-Type': 'application/json'}
+    clone_request_json = {'operation_type': 'clone_snapshot', 'table_uri': src_tbl_uri}
+    response = requests.post(PIXELTABLE_API_URL, json=clone_request_json, headers=headers_json)
+    if response.status_code != 200:
+        raise excs.Error(f'Error cloning snapshot: {response.text}')
+    response_json = response.json()
+    if not isinstance(response_json, dict) or 'table_uri' not in response_json:
+        raise excs.Error(f'Error cloning shapshot: unexpected response from server.\n{response_json}')
+
+    print(response_json)
+    primary_tbl_additional_md = response_json['md']['tables'][0]['table_md']['additional_md']
```

Comment:
Does this have a declared structure somewhere? As it is, it's hard to follow.

Reply:
Only in the Notion doc.

We should probably make it into a Pydantic structure with appropriate validators that's shared between the local/cloud logic. I've had some discussions with Amit about this.